test/e2e/l7networkpolicy_test.go

```diff
@@ -254,14 +280,61 @@ func testL7NetworkPolicyHTTP(t *testing.T, data *TestData) {
 		// will be rejected.
-		probeL7NetworkPolicy(t, data, serverPodName, clientPodName, serverIPs, true, false)
+		probeL7NetworkPolicyHTTP(t, data, serverPodName, clientPodName, serverIPs, true, false)
 
 		// Delete the first L7 NetworkPolicy that only allows HTTP path 'hostname'.
 		data.crdClient.CrdV1alpha1().NetworkPolicies(data.testNamespace).Delete(context.TODO(), policyAllowPathHostname, metav1.DeleteOptions{})
 		time.Sleep(networkPolicyDelay)
 
 		// Since the fist L7 NetworkPolicy has been deleted, corresponding packets will be matched by the second L7 NetworkPolicy,
 		// and the second L7 NetworkPolicy allows any HTTP path, then both path 'hostname' and 'clientip' are allowed.
-		probeL7NetworkPolicy(t, data, serverPodName, clientPodName, serverIPs, true, true)
+		probeL7NetworkPolicyHTTP(t, data, serverPodName, clientPodName, serverIPs, true, true)
 	})
 }
+
+func testL7NetworkPolicyTLS(t *testing.T, data *TestData) {
+	clientPodName := "test-l7-tls-client-selected"
+	clientPodLabels := map[string]string{"test-l7-tls-e2e": "client"}
+	cmd := []string{"bash", "-c", "sleep 3600"}
+
+	// Create a client Pod which will be selected by test L7 NetworkPolices.
+	require.NoError(t, NewPodBuilder(clientPodName, data.testNamespace, agnhostImage).OnNode(nodeName(0)).WithCommand(cmd).WithLabels(clientPodLabels).Create(data))
+	if _, err := data.podWaitForIPs(defaultTimeout, clientPodName, data.testNamespace); err != nil {
+		t.Fatalf("Error when waiting for IP for Pod '%s': %v", clientPodName, err)
+	}
+	require.NoError(t, data.podWaitForRunning(defaultTimeout, clientPodName, data.testNamespace))
+
+	l7ProtocolAllowsGoogle := []crdv1alpha1.L7Protocol{
+		{
+			TLS: &crdv1alpha1.TLSProtocol{
+				SNI: "*.google.com",
+			},
+		},
+	}
+	l7ProtocolAllowsFacebook := []crdv1alpha1.L7Protocol{
+		{
+			TLS: &crdv1alpha1.TLSProtocol{
+				SNI: "*.facebook.com",
+			},
+		},
+	}
+
+	policyAllowSNI1 := "test-l7-tls-allow-sni-google"
+	policyAllowSNI2 := "test-l7-tls-allow-sni-facebook"
+
+	// Create two L7 NetworkPolicies, one allows TLS service name 'google.com', the other allows 'facebook.com'. Note
+	// that, the priority of the first one is higher than the second one, and they have the same appliedTo labels and Pod
+	// selector labels.
+	createL7NetworkPolicy(t, data, false, policyAllowSNI1, 1, nil, clientPodLabels, ProtocolTCP, 443, l7ProtocolAllowsGoogle)
+	createL7NetworkPolicy(t, data, false, policyAllowSNI2, 2, nil, clientPodLabels, ProtocolTCP, 443, l7ProtocolAllowsFacebook)
+	time.Sleep(networkPolicyDelay)
+
+	probeL7NetworkPolicyTLS(t, data, clientPodName, "www.google.com", true)
+	probeL7NetworkPolicyTLS(t, data, clientPodName, "www.facebook.com", false)
+
+	// Delete the first L7 NetworkPolicy that allows TLS server name 'google.com'.
+	data.crdClient.CrdV1alpha1().NetworkPolicies(data.testNamespace).Delete(context.TODO(), policyAllowSNI1, metav1.DeleteOptions{})
+	time.Sleep(networkPolicyDelay)
+
```

Comment:
Check www.google.com is not reachable now.

Reply:
Changed to `apis.google.com`.